### ISSUE TITLE
Add more methods for outlier identification

### ISSUE THREAD

**Issue by danibene** [neuropsychology/NeuroKit#646]
As mentioned in [this issue](https://github.com/neuropsychology/NeuroKit/issues/541), to deal with noisy data, there needs to be a way to determine which parts of the data need to be discarded.

**Describe the solution you'd like**
In the [find_outliers](https://github.com/neuropsychology/NeuroKit/blob/master/neurokit2/misc/find_outliers.py) function, there is a single method for outlier identification using the z-score, but other more robust methods might work better in practice.

**How could we do it?**
[Here](https://github.com/danibene/NeuroKit/blob/feature/expand_outlier_identification/neurokit2/misc/find_outliers.py) I added another outlier identification method based on percentiles. 

Other methods (e.g. from [MATLAB](https://www.mathworks.com/help/matlab/ref/isoutlier.html)) could be added too and compared in terms of their performance on detecting distorted parts of signals.

Let me know if you're interested/have any feedback :-)

**Comment by DominiqueMakowski** [comment]
Yes, that's a good idea! Would you mind contributing this addition and opening a PR? thanks :)
